pagopa/io-app
# feat(IT Wallet): [SIW-326] Add WIA issuing flow

## Short description
This PR adds the wallet instance attestation issuing flow by integrating `io-react-native-wallet`. 
The app interacts with the keychain to create a pair of hardware-backed cypto keys to request a WIA.

## List of changes proposed in this pull request
- `.env.local && .env.production`: Adds a new flag for the wallet provider URL;
- `package.json && yarn.lock`: Updates `io-react-native-wallet` version;
- `ts/config.ts`: Exports the new wallet provider URL flag;
- `ts/features/it-wallet/saga/__tests__/handleWiaRequest.ts`: Updates tests for the WIA request saga;
- `ts/features/it-wallet/saga/handleWiaRequest.ts`: Updates the old requirements saga which now checks for requirements and then tries to obtain a WIA. 
The NFC check has been simplified by using already defined utility functions. 
- `ts/features/it-wallet/saga/index.ts`: Updates naming; 
- `ts/features/it-wallet/screens/issuing/ItwActivationInfoAuthScreen.tsx`: Updates naming;
- `ts/features/it-wallet/store/actions/index.ts`: Updates the actions to dispatch a `string` on success; 
- `ts/features/it-wallet/store/reducers/index.ts`: Updates naming; 
- `ts/features/it-wallet/store/reducers/itwWia.ts`: Updates naming;
- `ts/features/it-wallet/utils/errors/itwErrors.ts`: Adds a new error for the WIA issuing flow (not yet mapped to UI strings);
- `ts/features/it-wallet/utils/wia.ts`: Adds two utility functions to get the public key from the keychain and to request a WIA;

## How to test
Start the app and try to active the wallet. Using flipper, the WIA state should be filled properly and the activation should be successful.
